pkg/controller/status/status.go

```diff
@@ -301,6 +302,17 @@ func (m *statusManager) syncState() bool {
 	defer m.lock.Unlock()
 	progressing := []string{}
 	failing := []string{}
+
+	// Remove status displayed for CR, if CR not found.
+	if !m.enabled {
```

Comment:
Initially I had the cleanup code at the end of `Run()`, but the flow never goes beyond `syncState` as all the resources are cleaned up when `enabled` is set to false in `onCRNotFound`. Flow gets stuck in this loop https://github.com/tigera/operator/blob/master/pkg/controller/status/status.go#L108

How about calling crCleanup twice, once at the end of `syncState` when we have no more resources to watch and once at the end of `Run()`? Though I am not sure if the one at the end of `Run()` will ever get called, since we cleanup resource when CR not found.

Reply:
I'm still not a fan of syncState having that logic, this isn't "syncing the state" of the resources. You could run it in that Run()'s loop but regardless of the outcome of syncState

Reply:
True, I've moved it out.